Common/Subworlds/MappingAreas/ForestArea.cs

```diff
@@ -0,0 +1,475 @@
+﻿using PathOfTerraria.Common.ItemDropping;
+using PathOfTerraria.Common.Subworlds.BossDomains;
+using PathOfTerraria.Common.World;
+using PathOfTerraria.Common.World.Generation;
+using PathOfTerraria.Content.NPCs.Mapping.Forest.GrovetenderBoss;
+using PathOfTerraria.Content.Projectiles.Utility;
+using PathOfTerraria.Content.Tiles.Maps.Forest;
+using PathOfTerraria.Core.Items;
+using SubworldLibrary;
+using System.Collections.Generic;
+using Terraria.DataStructures;
+using Terraria.GameContent.Generation;
+using Terraria.ID;
+using Terraria.IO;
+using Terraria.Localization;
+using Terraria.WorldBuilding;
+
+namespace PathOfTerraria.Common.Subworlds.MappingAreas;
+
+internal class ForestArea : MappingWorld
+{
+	private enum StructureKind : byte
+	{
+		Cave,
+		Shrine,
+		Campsite,
+		AncientTree,
+		Arena,
+		Count,
+	}
+
+	public const int FloorY = 180;
+
+	private static bool LeftSpawn = false;
+	private static int LastTreeX = 0;
+	private static Point BossSpawnLocation = Point.Zero;
+
+	public override int Width => 1200;
+	public override int Height => 290;
```

Comment:
Are these things able to be dynamic? I feel like that could be fun.
Something like a range that could be set so the size is dynamic as well.